src/api.ts

```diff
@@ -0,0 +1,70 @@
+import { request } from "./common"
+
+export type SpawnRequestBody = {
+    env?: Record<string, string>; // env vars always map strings to strings
+    port?: number;
+    tag?: string;
+}
+
+// TODO(paul): formalize the API return types and create a public API spec (OpenAPI?)
+
+export class JamsocketApi {
```

Comment:
Not a huge deal, but I wonder if we need a class here instead of just exporting pure functions. Most places its usage goes like this:

```js
const api = new JamsocketApi(auth)
api.someAction()
```

It seems like it just functions as a place to store the `auth` value for basically one line and then have it used implicitly in the call on the next line. Maybe all this would be simpler if each function just explicitly took `auth` as an argument (i.e. `someAction(auth)`). In my opinion, it would make the logic more explicit and allow us to delete some more code.

Reply:
I'd still like to wire up an option to print out a `curl` command in the near future. I haven't worked out how it should work yet (e.g. does it just log when it runs, or does it act as a dry run), but I have a feeling that having `JamsocketApi` be a class will come in handy for that. If it turns out not to be, I'm on board for removing the class.